gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/IdentityRemovalStrategy.java

```diff
@@ -65,8 +66,12 @@ public void apply(final Traversal.Admin<?, ?> traversal) {
                 // gets to be:
                 // [UnionStep([[VertexStep(OUT,vertex), EndStep], [IdentityStep, EndStep], [VertexStep(OUT,vertex), EndStep]])]
                 // EndStep seems to just behave like a identity() in the above case, but perhaps it is more consistent
-                // to keep the identity() placeholder rather than a step that doesn't actually exist
-                if (!(identityStep.getNextStep() instanceof ComputerAwareStep.EndStep && traversal.getSteps().size() == 2)) {
+                // to keep the identity() placeholder rather than a step that doesn't actually exist.
```

Comment:
Nit: Would you mind making a quick update to the javadoc comment at the top of this class to mention the EndStep and RepeatEndStep exceptions? That comment looks a bit out of date.